src/Umbraco.Infrastructure/Persistence/Repositories/Implement/DocumentVersionRepository.cs

```diff
@@ -186,6 +180,31 @@ public void SetPreventCleanup(int versionId, bool preventCleanup)
             .On<UserDto, ContentVersionDto>(left => left.Id, right => right.UserId)
             .Where<ContentVersionDto>(x => x.Id == versionId);
 
-        return _scopeAccessor.AmbientScope?.Database.Single<ContentVersionMeta>(query);
+        ContentVersionMeta? result = _scopeAccessor.AmbientScope?.Database.Single<ContentVersionMeta>(query);
+        EnsureUtcDate(result);
+        return result;
+    }
+
+    private static void EnsureUtcDate(ContentVersionMeta? version)
+    {
+        if (version is null)
+        {
+            return;
+        }
+
+        EnsureUtcDates([version]);
+    }
+
+    private static void EnsureUtcDates(IEnumerable<ContentVersionMeta>? versions)
+    {
+        if (versions is null)
+        {
+            return;
+        }
+
+        foreach (ContentVersionMeta version in versions)
+        {
+            version.EnsureUtc();
+        }
     }
```

Comment:
I prefer to do `?? []` (same as `Enumerable.Empty<ContentVersionMeta>()`) on the foreach instead of having those few extra null check lines, but it's a preference.
```suggestion
    private static void EnsureUtcDates(IEnumerable<ContentVersionMeta>? versions)
    {
        foreach (ContentVersionMeta version in versions ?? [])
        {
            version.EnsureUtc();
        }
    }
```